src/SourceGenerators/Uno.UI.Tasks/Content/Uno.UI.Tasks.targets

```diff
@@ -178,66 +180,113 @@
+		Condition="'$(DesignTimeBuild)' != 'true'"
+		AfterTargets="UnoResourcesGeneration">
+
+		<ItemGroup>
+			<_UnoUpriResource Condition="'%(GeneratedFiles.UnoResourceTarget)' == 'Uno'" Include="@(GeneratedFiles->'%(LogicalName)')">
+				<Language>%(GeneratedFiles.Language)</Language>
+			</_UnoUpriResource>
+		</ItemGroup>
+
+		<UpriSubstitutionsGeneratorTask_v0
+			AssemblyName="$(AssemblyName)"
+			Resources="@(_UnoUpriResource)"
+			OutputFile="$(IntermediateOutputPath)\Substitutions\UpriResources.Substitutions.xml" />
+
+		<ItemGroup>
+			<FileWrites Include="$(IntermediateOutputPath)\Substitutions\UpriResources.Substitutions.xml" />
+		</ItemGroup>
+
+	</Target>
+
+	<Target Name="_UnoLinkerUpriFeatures"
+		Condition="'$(IsUnoHead)'=='true'"
+		BeforeTargets="BuildDist">
+
+		<UpriFeaturesGeneratorTask_v0
+			Languages="@(UnoSupportedLanguage)">
+			<Output TaskParameter="OutputFeatures" ItemName="_UpriOutputFeatures" />
+		</UpriFeaturesGeneratorTask_v0>
+
+		<ItemGroup>
+			<RuntimeHostConfigurationOption Include="@(_UpriOutputFeatures)" Trim="true" />
+		</ItemGroup>
+
+	</Target>
+
+	<!--
+	  Linker Substitutions Merge
 	  -->
 
-	<Target Name="_UnoLinkerHintsSubstitutionsMerge"
-		Condition="
-		'$(DesignTimeBuild)' != 'true'
-		and '$(BuildingInsideUnoSourceGenerator)' == ''
-		and '$(SkipCompilerExecution)' == ''
-		and '$(UnoXamlResourcesTrimming)'=='true'
-		and $(_IsXamlTrimmingAvailable)"
+	<Target Name="_UnoLinkerCollectSubstitutions"
+		Condition="'$(_UnoRequiresEmbeddedResourcesInjection)' == 'true'"
 		AfterTargets="CoreCompile">
 
 		<ItemGroup>
 			<_SubstitutionFiles Include="@(UnoLinkerSubstitution)" />
 			<_SubstitutionFiles Include="$(IntermediateOutputPath)\Substitutions\*.Substitutions.xml" />
 		</ItemGroup>
 
+	</Target>
+
+	<Target Name="_UnoLinkerSubstitutionsMerge"
+		Condition="'$(_UnoRequiresEmbeddedResourcesInjection)' == 'true' and @(_SubstitutionFiles->Count()) > 0"
+		DependsOnTargets="_UnoLinkerCollectSubstitutions"
+		BeforeTargets="_UnoEmbeddedResourcesInjection">
+
 		<LinkerDefinitionMergerTask_v0
 			DefinitionFiles="@(_SubstitutionFiles)"
-			TargetDefinitionFile="$(IntermediateOutputPath)ILLink.Substitutions.xml"
-		/>
+			TargetDefinitionFile="$(IntermediateOutputPath)ILLink.Substitutions.xml" />
 
 		<ItemGroup>
 			<EmbeddedResource Include="$(IntermediateOutputPath)ILLink.Substitutions.xml" LogicalName="ILLink.Substitutions.xml" />
-			
+
 			<FileWrites Include="$(IntermediateOutputPath)ILLink.Substitutions.xml" />
 		</ItemGroup>
 
 	</Target>
 
+	<PropertyGroup>
+		<TargetsTriggeredByCompilation>$(TargetsTriggeredByCompilation);_UnoSetupEmbeddedResourcesInjection</TargetsTriggeredByCompilation>
+	</PropertyGroup>
+
+	<Target Name="_UnoSetupEmbeddedResourcesInjection">
+
+		<PropertyGroup Condition="'$(DesignTimeBuild)' != 'true' and '$(SkipCompilerExecution)' == ''">
+			<_UnoRequiresEmbeddedResourcesInjection>true</_UnoRequiresEmbeddedResourcesInjection>
```

Comment:
`_IsXamlTrimmingAvailable` check was dropped. Is it intentional?

Reply:
Also the `BuildingInsideUnoSourceGenerator` condition was dropped. This may have an effect on sync generator (not sure if the effect is only performance or if it's behavioral)